src/opm/parser/eclipse/EclipseState/Aquifer/NumericalAquifer/NumericalAquiferConnection.cpp

```diff
@@ -0,0 +1,133 @@
+    NumericalAquiferConnection::generateConnections(const Deck &deck, const EclipseGrid &grid)
+    {
+        std::map<size_t, std::map<size_t, NumericalAquiferConnection>> connections;
+        using AQUCON=ParserKeywords::AQUCON;
+        if ( !deck.hasKeyword<AQUCON>() ) return connections;
+
+        const auto& aqucon_keywords = deck.getKeywordList<AQUCON>();
+        for (const auto& keyword : aqucon_keywords) {
+            OpmLog::info(OpmInputError::format("Initializing numerical aquifer connections from {keyword} in {file} line {line}", keyword->location()));
+            for (const auto& record : *keyword) {
+                const auto cons_from_record = NumericalAquiferConnection::connectionsFromSingleRecord(grid, record);
+                for (auto con : cons_from_record) {
+                    const size_t aqu_id = con.aquifer_id;
+                    const size_t global_index = con.global_index;
+                    auto& aqu_cons = connections[aqu_id];
+                    if (aqu_cons.find(global_index) == aqu_cons.end()) {
+                        aqu_cons.insert({global_index, con});
+                    } else {
+                        auto error = fmt::format("Numerical aquifer cell at ({}, {}, {}) is declared more than once"
+                                                 " as a connection for numerical aquifer {}",
+                                                 con.I + 1, con.J + 1, con.K + 1, con.aquifer_id);
+                        throw OpmInputError(error, keyword->location());
+                    }
+                }
+            }
+        }
+        return connections;
+    }
+
+    // TODO: we should not need all following the information here
+    using AQUCON = ParserKeywords::AQUCON;
+    NumericalAquiferConnection::NumericalAquiferConnection(const size_t i, const size_t j, const size_t k,
+                                                           const size_t global_index_in, const bool allow_connection_active, const DeckRecord& record)
+    : aquifer_id(record.getItem<AQUCON::ID>().get<int>(0))
+    , I(i)
+    , J(j)
+    , K(k)
+    , global_index(global_index_in)
+    , face_dir(FaceDir::FromString(record.getItem<AQUCON::CONNECT_FACE>().getTrimmedString(0)))
+    , trans_multipler(record.getItem<AQUCON::TRANS_MULT>().get<double>(0))
+    , trans_option(record.getItem<AQUCON::TRANS_OPTION>().get<int>(0))
+    , connect_active_cell(allow_connection_active)
+    , ve_frac_relperm(record.getItem<AQUCON::VEFRAC>().get<double>(0))
+    , ve_frac_cappress(record.getItem<AQUCON::VEFRACP>().get<double>(0))
+    {
+    }
+
+    std::vector<NumericalAquiferConnection> NumericalAquiferConnection::connectionsFromSingleRecord(const EclipseGrid& grid, const DeckRecord& record) {
+        std::vector<NumericalAquiferConnection> cons;
+
+        const size_t i1 = record.getItem<AQUCON::I1>().get<int>(0) - 1;
+        const size_t j1 = record.getItem<AQUCON::J1>().get<int>(0) -1;
```

Comment:
That alignment tickles my OCD

Reply:
if you are referring to `-1;`, I fixed it just now.